docs/design/w3c/spec.md

```diff
@@ -0,0 +1,530 @@
+* `revocation` - id of Indy Revocation Registry
+* `encoding` - attributes encoding algorithm
+    * encoded credential attribute values (binary representation required for doing CL signatures) are not included
+      neither to `credentialSubject` or `signature`
+    * `encoding: auto` implies using the algorithm defined
+      at [Aries RFC 0592 Indy Attachments section](https://github.com/hyperledger/aries-rfcs/tree/main/features/0592-indy-attachments#encoding-claims)
+      to generate encoded values under the hood during the signature generation and proof verification.
+
+#### Issuer
+
+W3C [Issuer](https://www.w3.org/TR/vc-data-model/#issuer) section requires including of `issuer` property to express the
+issuer of a verifiable credential.
+
+In the case of W3C AnonCreds credentials, the `issuer` attribute should be represented as a
+resolvable [DID URL](https://w3c-ccg.github.io/did-resolution/) having either `indy` or `sov` DID method and looks the
+following:
+
+```
+{
+  ... 
+  "issuer": "did:sov:3avoBCqDMFHFaKUHug9s8W",
+  ...
+}
+```
+
+#### Issuance Date
+
+W3C [Issuance Date](https://www.w3.org/TR/vc-data-model/#issuance-date) section requires including of `issuanceDate`
+property to express the date and time when a credential becomes valid.
+
+In the case of W3C AnonCreds credentials, for the `issuanceDate` attribute recommended setting of a random time of the
+day when credential was issued or transformed and looks the following:
+
+```
+{
+  ... 
+  "issuanceDate": "2010-01-01T19:23:24Z",
+  ...
+}
+```
+
+#### Proofs (Signatures)
+
+W3C [Proofs (Signatures)](https://www.w3.org/TR/vc-data-model/#proofs-signatures) section requires including of `proof`
+property to express confirmation of the credential's validity.
+
+According to the specification, one or many proof objects can be added to verifiable credentials.
+In the context of W3C AnonCreds credentials included at least two proof object entries: AnonCreds CL
+and [Data Integrity](https://www.w3.org/TR/vc-data-model/#data-integrity-proofs).
+
+##### AnonCreds CL proof
+
+This proof entry derived from the CL signature of a verifiable credential.
+
+The defined `@context` includes a definition for the `CLSignature2022` type describing the format of the proof
+entry:
+
+```
+{
+  ... 
+  "proof": [
+    {
+      "type": "CLSignature2022",
+      "signature": "AAAgf9w5lZg....RYp8Z_x3FqdwRHoWruiF0FlM"
+    }
+  ]  
+  ...
+}
+```
+
+**Credential proof signature**
+
+* `type` - `CLSignature2022`
+* `signature` - credential signature
+    * signature received by building the following object from indy styled credential:
+      ```
+      {
+        "signature": {..}, 
+        "signature_correctness_proof": {..}
+      }
+      ```
+    * encoded
+      as [base64 attachment](https://github.com/hyperledger/aries-rfcs/tree/main/concepts/0017-attachments#base64url)
+
+
+* **TO DISCUSS**: Signature/Proof encoding: Which approach to use for encoding?
```

Comment:
I think we should really see the size difference here and how much it would add to an implementer to support this custom compact encoding. 

If the size difference is negligible or it is rather complex, I'd rather sick with base64url. If it is significant or trivial to implement the compact would be my preference.

Reply:
I think the more complex encoding is fine, vs. straight serialized JSON encoding.  But agree that seeing the size difference would be the right thing to understand the tradeoff.